url73e3057d61_empeeoh_BACnet4J_tgz-pJ8-com_serotonin_bacnet4j_test_OverflowTestJ8/src/com/serotonin/bacnet4j/LocalDevice.java

```diff
@@ -0,0 +1,648 @@
+
+            // Set up the object types supported.
+            ObjectTypesSupported objectTypesSupported = new ObjectTypesSupported();
+            objectTypesSupported.setAll(true);
+            configuration.setProperty(PropertyIdentifier.protocolObjectTypesSupported, objectTypesSupported);
+
+            // Set some other required values to defaults
+            configuration.setProperty(PropertyIdentifier.objectName, new CharacterString("BACnet device"));
+            configuration.setProperty(PropertyIdentifier.systemStatus, DeviceStatus.operational);
+            configuration.setProperty(PropertyIdentifier.modelName, new CharacterString("BACnet4J"));
+            configuration.setProperty(PropertyIdentifier.firmwareRevision, new CharacterString("not set"));
+            configuration.setProperty(PropertyIdentifier.applicationSoftwareVersion, new CharacterString("1.0.1"));
+            configuration.setProperty(PropertyIdentifier.protocolVersion, new UnsignedInteger(1));
+            configuration.setProperty(PropertyIdentifier.protocolRevision, new UnsignedInteger(0));
+            configuration.setProperty(PropertyIdentifier.databaseRevision, new UnsignedInteger(0));
+        }
+        catch (BACnetServiceException e) {
+            // Should never happen, but wrap in an unchecked just in case.
+            throw new BACnetRuntimeException(e);
+        }
+    }
+
+    public ExecutorService getExecutorService() {
+        return executorService;
+    }
+
+    public void setExecutorService(ExecutorService executorService) {
+        if (initialized)
+            throw new IllegalStateException("Cannot set the executor service. Already initialized");
+        this.executorService = executorService;
+    }
+
+    public NetworkIdentifier getNetworkIdentifier() {
+        return transport.getNetworkIdentifier();
+    }
+
+    /**
+     * @return the strict
+     */
+    public boolean isStrict() {
+        return strict;
+    }
+
+    /**
+     * @param strict
+     *            the strict to set
+     */
+    public void setStrict(boolean strict) {
+        this.strict = strict;
+    }
+
+    public synchronized void initialize() throws Exception {
+        if (executorService == null) {
+            executorService = Executors.newCachedThreadPool();
+            ownsExecutorService = true;
+        }
+        else
+            ownsExecutorService = false;
+        transport.initialize();
+        initialized = true;
+    }
+
+    public synchronized void terminate() {
+        transport.terminate();
+        initialized = false;
+
+        if (ownsExecutorService) {
+            ExecutorService temp = executorService;
+            executorService = null;
+            if (temp != null) {
+                temp.shutdown();
+                try {
+                    temp.awaitTermination(3, TimeUnit.SECONDS);
+                }
+                catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)